internal/app/proxyd/internal/frontend/frontend.go

```diff
@@ -259,14 +259,11 @@ func (r *ReverseProxy) proxyConnection(c1 net.Conn) {
 		return
 	}
 
-	uid := backend.UID
-	addr := backend.Addr
-
-	c2, err := net.DialTimeout("tcp", addr+":6443", time.Duration(r.ConnectTimeout)*time.Millisecond)
+	c2, err := net.DialTimeout("tcp", backend.Addr+":6443", time.Duration(r.ConnectTimeout)*time.Millisecond)
 	if err != nil {
-		log.Printf("dial %v: %v", addr, err)
-		// nolint: errcheck
-		c1.Close()
```

Comment:
Pretty sure if this works like I think, we're going to run this down till we get a successful connection to a backend, or we hit line 254 which should close it for us. 
```
	backend := r.GetBackend()
	if backend == nil {
		log.Printf("no available backend, closing remote connection: %s", c1.RemoteAddr().String())
		// nolint: errcheck
		c1.Close()
			return
		}
```

Reply:
Yep looks like it to me.